plugins/processors/gpuattributes/processor.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package gpuattributes
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/aws/amazon-cloudwatch-agent/internal/containerinsightscommon"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+const (
+	gpuMetricIdentifier      = "_gpu_"
+	gpuContainerMetricPrefix = "container_"
+	gpuPodMetricPrefix       = "pod_"
+	gpuNodeMetricPrefix      = "node_"
+)
+
+// schemas at each resource level
+// - Container Schema
+//   - ClusterName
+//   - ClusterName, Namespace, PodName, ContainerName
+//   - ClusterName, Namespace, PodName, FullPodName, ContainerName
+//   - ClusterName, Namespace, PodName, FullPodName, ContainerName, GpuDevice
+//
+// - Pod
+//   - ClusterName
+//   - ClusterName, Namespace
+//   - ClusterName, Namespace, Service
+//   - ClusterName, Namespace, PodName
+//   - ClusterName, Namespace, PodName, FullPodName
+//   - ClusterName, Namespace, PodName, FullPodName, GpuDevice
+//
+// - Node
+//   - ClusterName
+//   - ClusterName, InstanceIdKey, NodeName
+//   - ClusterName, InstanceIdKey, NodeName, GpuDevice
+
+var commonLabels = []string{
+	containerinsightscommon.ClusterNameKey,
+	containerinsightscommon.InstanceIdKey,
+	containerinsightscommon.GpuDeviceKey,
+	containerinsightscommon.MetricType,
+	containerinsightscommon.NodeNameKey,
+	containerinsightscommon.VersionKey,
+	containerinsightscommon.SourcesKey,
+	containerinsightscommon.Timestamp,
+}
+
+var podAndContainerLabels = []string{
+	containerinsightscommon.K8sNamespace,
+	containerinsightscommon.FullPodNameKey,
+	containerinsightscommon.PodNameKey,
+	containerinsightscommon.TypeService,
+	containerinsightscommon.GpuUniqueId,
+}
+
+var containerK8sBlobLabels = []string{
```

Comment:
What are the extra "kubernetes" blob fields that you want to get rid of?

Reply:
Not really filtering out anything at container level, but listing all in the array to keep what's populated by the decorator.